tests/dependency_analyzer/cyclonedx/test_cyclonedx.py

```diff
@@ -6,26 +6,23 @@
 from pathlib import Path
 
 import pytest
+from cyclonedx.model.component import Component as CDXComponent
 
 from macaron.config.defaults import defaults, load_defaults
-from macaron.dependency_analyzer.cyclonedx import (
-    CycloneDXParserError,
-    convert_components_to_artifacts,
-    deserialize_bom_json,
-    get_dep_components,
-    get_deps_from_sbom,
-)
+from macaron.database.table_definitions import Analysis, Component, Repository
+from macaron.dependency_analyzer.cyclonedx import CycloneDXParserError, DependencyInfo, deserialize_bom_json
 from macaron.dependency_analyzer.cyclonedx_mvn import CycloneDxMaven
-from macaron.dependency_analyzer.dependency_resolver import DependencyInfo
+from macaron.slsa_analyzer.build_tool.base_build_tool import BaseBuildTool
 
 RESOURCES_DIR = Path(__file__).parent.joinpath("resources")
 
 
-def test_deserialize_bom_json(snapshot: dict) -> None:
+def test_deserialize_bom_json(snapshot: list[str]) -> None:
     """Test deserializing a bom.json file."""
     # Deserialize a valid bom.json.
     path = Path(RESOURCES_DIR, "valid_bom.json")
-    assert snapshot == deserialize_bom_json(path)
+    bom = deserialize_bom_json(path)
+    assert snapshot == [str(cmp.bom_ref) for cmp in bom.components if isinstance(cmp, CDXComponent)]
```

Comment:
I wonder why we moved away from testing the whole `bom` object, when we are de-serializing a static file that doesn't change. Is is intended to let this test pass, even in cases where the `deserialize_bom_json` method introduces other objects that is not type `CDXComponent` into the returned `bom` object?

Reply:
I think it would be good to have a small comment here about this issue that you encountered.

Reply:
See commit de28bafbf82894f10d157bbfaccaf65d1a6ecd83